manticore/platforms/linux.py

```diff
@@ -383,11 +393,18 @@ def _execve(self, program, argv, envp):
             self.forward_events_from(proc)
 
     def _mk_proc(self, arch):
-        if arch in {'i386', 'armv7'}:
-            mem = Memory32()
-        else:
-            mem = Memory64()
-        return CpuFactory.get_cpu(mem, arch)
+        mem = Memory32() if arch in {'i386', 'armv7'} else Memory64()
+        cpu = CpuFactory.get_cpu(mem, arch)
+        # FIXME
+        arch_map = {
+            'i386': (cs.CS_ARCH_X86, cs.CS_MODE_32),
+            'amd64': (cs.CS_ARCH_X86, cs.CS_MODE_64),
+            'armv7': (cs.CS_ARCH_ARM, cs.CS_MODE_ARM)
+        }
+        arch, mode = arch_map[arch]
+        cpu.__class__.disasm = init_disassembler(self.disasm, arch, mode, self.programs)
```

Comment:
Would it be possible to initialize the disasm in the cpu ctor? In practice, I think we will only have one disasm per cpu, so would this be fine?